### PR TITLE
Mb sdk module split

### PR DESCRIPTION
**Pull request by volkodavs**
Major update in code structure 

1) split core module into sub modules 
2) change name convention for class name (e.g. ventsRestClientImpl- > EventsClientRest, EventsRestClient -> EventsClient )

### CODE REVIEW

File: matchbook-sdk-common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>matchbook-sdk</artifactId>
+        <groupId>com.matchbook</groupId>
+        <version>0.1.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>matchbook-sdk-common</artifactId>
```
Comment (alessandrodalbello) [inline matchbook-sdk-common/pom.xml]: I don't like _common_ very much. What about _api_? Or maybe _core_ and the current _core_ renamed to _rest_?

File: matchbook-sdk-stream/pom.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>matchbook-sdk</artifactId>
+        <groupId>com.matchbook</groupId>
+        <version>0.1.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>matchbook-sdk-stream</artifactId>
```
Comment (alessandrodalbello) [inline matchbook-sdk-stream/pom.xml]: Also the name _stream_ I'm not sure is the best one...

File: matchbook-sdk-core/src/main/java/com/matchbook/sdk/core/dtos/user/LogoutRequest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.matchbook.sdk.core.dtos.user;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.matchbook.sdk.core.dtos.RestRequest;
+
+public class LogoutRequest implements RestRequest {
+
+    private LogoutRequest(LogoutRequest.Builder builder) {
```
Comment (alessandrodalbello) [inline matchbook-sdk-core/src/main/java/com/matchbook/sdk/core/dtos/user/LogoutRequest.java]: As pointed out also by Codacy, the builder is not used inside the constructor.

File: matchbook-sdk-common/pom.xml (hunk 1 above)
Comment (volkodavs) [reply matchbook-sdk-common/pom.xml]: yeah, you right. 
I would recommend you then to take a look on that split as on a logical structure, not a the names

File: matchbook-sdk-core/src/test/java/com/matchbook/sdk/core/clients/rest/EventClientImplTest.java [hunk 4]
```diff
@@ -11,28 +11,30 @@
...
+import com.matchbook.sdk.core.dtos.events.Runner;
+import com.matchbook.sdk.core.dtos.events.RunnerRequest;
+import com.matchbook.sdk.core.dtos.events.RunnersRequest;
+import com.matchbook.sdk.core.dtos.events.Sport;
+import com.matchbook.sdk.core.dtos.events.SportsRequest;
+import com.matchbook.sdk.common.exceptions.MatchbookSDKException;
 import org.junit.Test;
 
-public class EventRestClientImplTest extends MatchbookSDKClientTest {
+public class EventClientImplTest extends MatchbookSDKClientTest {
```
Comment (matteodri) [inline matchbook-sdk-core/src/test/java/com/matchbook/sdk/core/clients/rest/EventClientImplTest.java]: This class should be named _**EventsClientRestTest**_ as the unit being tested is a _EventsClientRest_ object

**Pull request merged**